pkg/autoscaler/stats_scraper.go

```diff
@@ -142,22 +144,48 @@ func (s *ServiceScraper) Scrape() (*StatMessage, error) {
 	sampleSize := populationMeanSampleSize(readyPodsCount)
 	statCh := make(chan *Stat, sampleSize)
 
-	url := s.target()
+	scrapedPods := sets.NewString()
+	var scrapedPodMux sync.Mutex
+	writeIfAbsent := func(name string) bool {
+		scrapedPodMux.Lock()
+		defer scrapedPodMux.Unlock()
+
+		if !scrapedPods.Has(name) {
+			scrapedPods.Insert(name)
+			return true
+		}
+		return false
+	}
+
 	grp := errgroup.Group{}
 	for i := 0; i < sampleSize; i++ {
 		grp.Go(func() error {
-			stat, err := s.sClient.Scrape(url)
-			if err != nil {
-				return err
+			var statToWrite *Stat
+			var tries int
+			for statToWrite == nil {
+				stat, err := s.sClient.Scrape(s.target())
```

Comment:
Do we want to retry on errors?

There could be some cases an error is returned from `Scrape`:
1. A pod just starts within 1 second and no metrics are reported(so the required keys are missing in Prometheus output).
1. The queue is in unhealthy stats, for example OOMKilled.

The first case happens frequently especially when scaling from small size, i.e. 1 to 2, 2 to 3.

Reply:
Yep, we can probably retain the error and return it if all iterations failed.